Sources/Implementation/DefaultDecisionService.swift

```diff
@@ -190,42 +212,101 @@ class DefaultDecisionService: OPTDecisionService {
         return DecisionResponse(result: result, reasons: reasons)
     }
     
+    /// Public Method
     func getVariationForFeature(config: ProjectConfig,
                                 featureFlag: FeatureFlag,
                                 user: OptimizelyUserContext,
                                 options: [OptimizelyDecideOption]? = nil) -> DecisionResponse<FeatureDecision> {
+        
+        let response = getVariationForFeatureList(config: config, featureFlags: [featureFlag], user: user, options: options).first
+        
+        guard response?.result != nil else {
+            let reasons = response?.reasons ?? DecisionReasons(options: options)
+            return DecisionResponse(result: nil, reasons: reasons)
+        }
+        
+        return response!
+        
+        
+        
+//        // Evaluate in this order:
+//        
+//        // 1. Attempt to bucket user into experiment using feature flag.
+//        // Check if the feature flag is under an experiment and the the user is bucketed into one of these experiments
+//        var decisionResponse = getVariationForFeatureExperiment(config: config,
+//                                                                featureFlag: featureFlag,
+//                                                                user: user,
+//                                                                userProfileTracker: nil,
+//                                                                options: options)
+//        reasons.merge(decisionResponse.reasons)
+//        if let decision = decisionResponse.result {
+//            return DecisionResponse(result: decision, reasons: reasons)
+//        }
+//        
+//        // 2. Attempt to bucket user into rollout using the feature flag.
+//        // Check if the feature flag has rollout and the user is bucketed into one of it's rules
+//        decisionResponse = getVariationForFeatureRollout(config: config,
+//                                                         featureFlag: featureFlag,
+//                                                         user: user,
+//                                                         options: options)
+//        reasons.merge(decisionResponse.reasons)
+//        if let decision = decisionResponse.result {
+//            return DecisionResponse(result: decision, reasons: reasons)
+//        }
+//        
+//        return DecisionResponse(result: nil, reasons: reasons)
+    }
+    
+    func getVariationForFeatureList(config: ProjectConfig,
+                                    featureFlags: [FeatureFlag],
+                                    user: OptimizelyUserContext,
+                                    options: [OptimizelyDecideOption]? = nil) -> [DecisionResponse<FeatureDecision>] {
```

Comment:
What is this extra function for? It looks doing same as the one above.

Reply:
This method is called by `decideForKeys` from [client](https://github.com/optimizely/swift-sdk/blob/e132c566b1607e4c9285a35f31be6f52830822c4/Sources/Optimizely%2BDecide/OptimizelyClient%2BDecide.swift#L131)